docs/developer/decision-records/2025-03-17-retry-processor-on-dataflow-completion/README.md

```diff
@@ -0,0 +1,45 @@
+# Retry Processor on Dataflow Completion
+
+## Decision
+
+The retry mechanism will be applied in the `DataFlowManagerImpl` when processing the `Dataflow`'s *COMPLETED* and *FAILED* states.
+
+## Rationale
+
+The data plane notifies the control plane of the success or failure of the data transfer. As of right now, if this notification fails, the `DataFlow` transitions to its current state, being picked up again by the state machine manager and retrying the process. As there is no limit to the amount of retries, this process could possibly continue forever.
+
+```java
+private boolean processCompleted(DataFlow dataFlow) {
+    var response = transferProcessClient.completed(dataFlow.toRequest());
+    if (response.succeeded()) {
+        dataFlow.transitToNotified();
+        update(dataFlow);
+    } else {
+        dataFlow.transitToCompleted(); // Will retry while the process fails
+        update(dataFlow);
+    }
+    return true;
+}
+
+private boolean processFailed(DataFlow dataFlow) {
+    var response = transferProcessClient.failed(dataFlow.toRequest(), dataFlow.getErrorDetail());
+    if (response.succeeded()) {
+        dataFlow.transitToNotified();
+        update(dataFlow);
+    } else {
+        dataFlow.transitToFailed(dataFlow.getErrorDetail()); // Will retry while the process fails
+        update(dataFlow);
+    }
+    return true;
+}
+```
+
+By adding a `RetryProcessor`, we may define handlers for when the process is successful, when it fails but can be retried, and when it reaches final failure.
+
+## Approach
+
```

Comment:
This discusses different code paths and the explanation is not clear. Please refine this to clearly demarcate which components and services are affected by the proposed change.